src/Controller/Admin/IndexController.php

```diff
@@ -224,6 +224,7 @@ protected function buildPimcoreSettings(
                 $systemSettings['general']['valid_languages'],
```

Comment:
```suggestion
                $systemSettings['general']['valid_languages'] ?? [],
```
@matthiashamacher this could be null now, since we removed the default value in https://github.com/pimcore/pimcore/pull/16835 for it, right?

Reply:
Not null as its defined to be an array. So it would always be an empty array. But with the logic change from pimcore/pimcore#16835 an empty array does not make any sense as then we would have no required language so I overwrite an empty array with the valid_languages as in pimcore/pimcore#16835

Reply:
I'm not quite sure how you got that error, I cannot reproduce it. I have removed the key from the `var/config/system_settings/system_settings.yaml` but still I cannot get that error.
Even though we remove the default value the key is still present in the config but as empty array. This is because of the prototype, which we have for the required_languages: https://github.com/pimcore/pimcore/pull/16835/files#diff-bab0da2e129502ead20a1f1ac979fd2260c5d4c8da202b6a8fbf1e9d906d4541R232
If you prototype an array node with scalar, then you get an empty array 
`->prototype('scalar')->end()` exists:
![CleanShot 2024-04-20 at 10 53 04](https://github.com/pimcore/admin-ui-classic-bundle/assets/32356549/7f9ac33b-e607-49ab-96d1-df8a462cad37)
`->prototype('scalar')->end()` removed:
![CleanShot 2024-04-20 at 10 53 18](https://github.com/pimcore/admin-ui-classic-bundle/assets/32356549/4dc7a7ec-d478-47ae-8639-d06e7a7d31c6)

Reply:
Aaah, ok, the problem is, that I am using the `settings-store` as a `write_target` for my system settings, not the symfony-config. 
Could you please try to change your `write_target` (https://pimcore.com/docs/platform/Pimcore/Tools_and_Features/System_Settings/) very briefly and see if you can reproduce the error?

Reply:
Even then I cannot reproduce that error...
Change `write_target` location:
![CleanShot 2024-04-22 at 08 54 48](https://github.com/pimcore/admin-ui-classic-bundle/assets/32356549/e8490cba-5151-4182-b908-c0d27bafd3ed)
Saved config in the `settings_store` table:
![CleanShot 2024-04-22 at 08 55 06](https://github.com/pimcore/admin-ui-classic-bundle/assets/32356549/b1caadb1-d23f-491b-8429-b6273e4d3850)
Still getting no error and on accessing the configs all checkboxes are set as if there is an empty array returned.

Reply:
Can we please give it one more shot? Could you please try to set both `write_target` and `read_target` to settings-store? 
Sorry for the troubles, but I was able to reproduce the issue on another machine/env too.

Reply:
Thank you for adding the check. 
I agree, currently the fallback is always `symfony-config`, but it would make more sense to set it to the value of `write_target`.